ILSwiss/run_scripts/gen_smarts_demos.py

```diff
@@ -194,6 +223,58 @@ def sample_demos(
         raw_observations = raw_next_observations
         observations = next_observations
 
+    print(f"worker process: {traffic_name} finished!")
+
+
+def sample_demos(
+    train_vehicle_ids,
+    scenarios,
+    obs_stack_size,
+    feature_list,
+    closest_neighbor_num,
+    use_rnn,
+):
+    scenario_iterator = Scenario.scenario_variations(
+        [scenarios], list([]), shuffle_scenarios=False, circular=False
+    )  # scenarios with different traffic histories.
+
+    worker_processes = []
+    trajs_queue = Queue()  # Queues are process safe.
+    for scenario in scenario_iterator:
+        traffic_name = scenario._traffic_history.name
+        if traffic_name not in train_vehicle_ids:
+            continue
+        p = Process(
+            target=work_process,
+            args=(
+                trajs_queue,
+                train_vehicle_ids[traffic_name],
+                scenario,
+                obs_stack_size,
+                feature_list,
+                closest_neighbor_num,
+                use_rnn,
+            ),
+            daemon=True,
+        )
+        print(f"Traffic {traffic_name} start sampling.")
+        p.start()
+        worker_processes.append(p)
+
+    # Don not call p.join().
+    demo_trajs = []
+    while True:  # not reliable
+        try:
+            if len(demo_trajs) == 0:
+                traj = trajs_queue.get(block=True)
+            else:
+                traj = trajs_queue.get(block=True, timeout=100)
+            demo_trajs.append(traj)
+            print(f"main process: collected trajs num: {len(demo_trajs)}")
+        except:
```

Comment:
LSTM suggests not using bare except to catch all exceptions, which may hide some other problems.
I will change this to only catch `queue.Empty` exception.